### PR TITLE
add dll to enable using winrt tts api in win32 

### PR DESCRIPTION
**Pull request by davidkline-ms**

### CODE REVIEW

File: WinRTTextToSpeech/WinRTTextToSpeech.sln [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.3.32929.385
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "WinRTTextToSpeech", "WinRTTextToSpeech\WinRTTextToSpeech.vcxproj", "{5CC3EDB7-5FFE-428A-8D25-73E88476433F}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|x64 = Debug|x64
+		Debug|x86 = Debug|x86
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech.sln:11]: Do we need x86? We don't ship any x86 binaries for MR OpenXR Plugin (and neither does Unity 😅).

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+/// <param name="buffer">The audio data in wave format.</param>
+/// <param name="bufferLength">The length of the data being returned.</param>
+/// <returns>True of the synthesis is successful, or false.</returns>
+DLLEXPORT bool __stdcall TrySynthesizePhrase(
+	const char* phrase,
+	BYTE** data,
+	uint32_t& dataLength)
+{
+	// Create the synthesizer on first use.
+	if (!synthesizer)
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:31]: nit: tabs 😬

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	{
+		synthesizer = Windows::Media::SpeechSynthesis::SpeechSynthesizer();
+
+		// Confirm one was created successfully.
+		if (!synthesizer)
+		{
+			data = NULL;
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:38]: nit: could all the `NULL` usage be `nullptr`?

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		dataLength = 0;
+		return false;
+	}
+
+	// Read the syntesized data and pass it back to the caller.
+	reader.LoadAsync(dataLength).get();
+	std::vector<uint8_t>* temp = new std::vector<uint8_t>(dataLength);
+	array_view<uint8_t>* tempView =new array_view<uint8_t>(*temp);
+	reader.ReadBytes(*tempView);
+	*data = temp->data();
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:78]: Is there anything here we need to explicitly free (`new` makes me think there might be...)? If not, is there any guarantee how long after this method returns that the `temp->data()` pointer will point to valid memory?

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+		inputStream.Close();
+		data = NULL;
+		dataLength = 0;
+		return false;
+	}
+
+	// Read the syntesized data and pass it back to the caller.
+	reader.LoadAsync(dataLength).get();
+	std::vector<uint8_t>* temp = new std::vector<uint8_t>(dataLength);
+	array_view<uint8_t>* tempView =new array_view<uint8_t>(*temp);
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:76]: Interesting...I thought `ReadBytes` could take in a `std::vector` directly. I assume I'm wrong on that?

Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:76]: Either way, though, does this need to be `array_view<uint8_t>*` or could it just be `array_view<uint8_t> tempView = array_view<uint8_t>(*temp);`? An extended/manual lifecycle for this seems unimportant. (or even more succinctly, could the next line just be `reader.ReadBytes(array_view<uint8_t>(*temp));`?

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+/// <summary>
+/// Synthesizes the provided text into wave format.
+/// </summary>
+/// <param name="phrase">The phrase to be synthesized.</param>
+/// <param name="buffer">The audio data in wave format.</param>
+/// <param name="bufferLength">The length of the data being returned.</param>
+/// <returns>True of the synthesis is successful, or false.</returns>
+DLLEXPORT bool __stdcall TrySynthesizePhrase(
+	const char* phrase,
+	BYTE** data,
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:27]: I don't know why, but ALL CAPS types in C++/WinRT always makes me feel uneasy haha 😅 could this just be `uint8_t**`, which is the type used below anyway? (`std::vector<uint8_t>* temp = new std::vector<uint8_t>(dataLength);`)

File: WinRTTextToSpeech/WinRTTextToSpeech.sln (hunk 1 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech.sln:11]: i'm good with x64 only.

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 2 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:31]: will fix.

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 3 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:38]: sure!

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 4 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:78]: we should be able to free at least the array_view. we should chat about the vector.

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 5 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:76]: when i tried that, it appeared to only want to take an array_view

Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:76]: let me give that a shot :smile:

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 6 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:27]: sure can!

File: WinRTTextToSpeech/WinRTTextToSpeech.sln (hunk 1 above)
Comment (MaxWang-MS) [reply WinRTTextToSpeech/WinRTTextToSpeech.sln:11]: Shall we add ARM and ARM64 as well?

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+SpeechSynthesizer synthesizer = nullptr;
+
+EXTERN_C
+{
+
+/// <summary>
+/// Synthesizes the provided text into wave format.
+/// </summary>
+/// <param name="phrase">The phrase to be synthesized.</param>
+/// <param name="buffer">The audio data in wave format.</param>
```
Comment (MaxWang-MS) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:22]: Two param names seems to be wrong

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+/// <param name="bufferLength">The length of the data being returned.</param>
+/// <returns>True of the synthesis is successful, or false.</returns>
```
Comment (MaxWang-MS) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:24]:
```suggestion
/// <returns>True if the synthesis is successful, or false.</returns>
```

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.vcxproj [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+      <SubSystem>Windows</SubSystem>
+      <EnableCOMDATFolding>true</EnableCOMDATFolding>
+      <OptimizeReferences>true</OptimizeReferences>
+      <GenerateDebugInformation>true</GenerateDebugInformation>
+      <EnableUAC>false</EnableUAC>
+    </Link>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ClInclude Include="pch.h" />
+    <ClInclude Include="WinRTTextToSpeech.h" />
```
Comment (MaxWang-MS) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.vcxproj:147]: This file does not seem to be checked in

Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.vcxproj:147]: ah yes... that one isnt needed :)

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 5 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp]: will double check, to be sure

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 4 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp]: array_view is no longet being new'd. the vector still needs to be or the data will not marshal correctly

Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp]: rewrote this section to simplify allocations and frees

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 5 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp]: rewrote to be nicer :smile:

File: WinRTTextToSpeech/UnityAddon/WinRTTextToSpeechPinvokes.cs [hunk 10]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#if (UNITY_STANDALONE_WIN || UNITY_EDITOR_WIN)
```
Comment (keveleigh) [inline WinRTTextToSpeech/UnityAddon/WinRTTextToSpeechPinvokes.cs:4]: Do "empty" asmdefs (like this one will be whenever this `#if` is false) still log something like "welp I'm empty but I guess I'll exist anyway" or is it silent now?

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+/// <summary>
+/// Synthesizes the provided text into wave format.
+/// </summary>
+/// <param name="phrase">The phrase to be synthesized.</param>
+/// <param name="data">Will receive the audio data in wave format.</param>
+/// <param name="dataLength">The length of the data being returned.</param>
+/// <returns>True if the synthesis is successful, or false.</returns>
+DLLEXPORT bool __stdcall TrySynthesizePhrase(
+	const char* phrase,
```
Comment (keveleigh) [inline WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:26]: nit: still looks like there are tabs here, but I'm not too bothered if you want to leave it as-is

**Review by keveleigh — APPROVED** [review]

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 11 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:26]: weird. i'll double check

File: WinRTTextToSpeech/UnityAddon/WinRTTextToSpeechPinvokes.cs (hunk 10 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/UnityAddon/WinRTTextToSpeechPinvokes.cs:4]: let me double check

File: WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp (hunk 11 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/WinRTTextToSpeech/WinRTTextToSpeech.cpp:26]: used vscode to confirm/redo the indentation (it shows spaces). git also does not see any change to the file. i think this one is ok now.

File: WinRTTextToSpeech/UnityAddon/WinRTTextToSpeechPinvokes.cs (hunk 10 above)
Comment (davidkline-ms) [reply WinRTTextToSpeech/UnityAddon/WinRTTextToSpeechPinvokes.cs:4]: tweaked the #if placement and added a "windows platform only" comment

**Pull request merged**